hydrolib/core/dflowfm/tim/serializer.py

```diff
@@ -0,0 +1,100 @@
+        """Serialize the provided timeserie with the given config
+
+        Args:
+            timeserie (TimTimeSerie): The timeserie to serialize
+            config (TimSerializerConfig): The config describing the serialization options
+
+        Returns:
+            Lines: The iterable lines of the serialized timeserie
+        """
+        datablock = self._create_tim_datablock(timeserie, config)
+        return self._tim_serialize_datablock(datablock)
+
+    def _create_tim_datablock(self, timeserie: TimTimeSerie, config) -> Datablock:
+        datablock = []
+        row = self._create_tim_data_row(timeserie, config.float_format)
+        datablock.append(row)
+        return datablock
+
+    def _create_tim_data_row(self, timeserie: TimTimeSerie, float_format: str):
+        format_float = lambda x: f"{x:{float_format}}"
+        series = [str(format_float(p)) for p in timeserie.series]
+        row = []
+        row.append(str(timeserie.time))
+        row.extend(series)
+        return row
+
+    def _tim_serialize_datablock(self, datablock: Optional[Datablock]) -> Lines:
+        if datablock is None:
+            return []
+
+        indent = " " * self._config.datablock_indent
+        return self._tim_serialize_row(datablock[0], indent)
+
+    def _tim_serialize_row(self, row: DatablockRow, indent: str) -> str:
+        elem_spacing = " " * self.config.datablock_spacing
+        elems = (self._tim_serialize_row_element(elem) for elem in row)
+        return indent + elem_spacing.join(elems).rstrip()
+
+    def _tim_serialize_row_element(self, elem: str) -> str:
+        whitespace = _tim_get_offset_whitespace(elem, self.MAX_LENGTH)
+        return elem + whitespace
+
+
+def _tim_get_offset_whitespace(key: Optional[str], max_length: int) -> str:
+    key_length = len(key) if key is not None else 0
+    return " " * max(max_length - key_length, 0)
+
+
+class TimSerializer:
+    @staticmethod
+    def serialize(
+        path: Path, data: List[TimTimeSerie], config: TimSerializerConfig
+    ) -> None:
+        """
+        Serializes the timeseries data to the file at the specified path in .tim format.
+
+        Attributes:
+            path (Path): The path to the destination file.
+            data (List): The data to be serialized.
+            config (TimSerializerConfig): The serialization configuration.
+        """
+        path.parent.mkdir(parents=True, exist_ok=True)
+        serializer = TimTimeSeriesSerializer(config, MaxLengths(key=0, value=0))
+
+        with path.open("w") as file:
+            for timeserie in data:
+                TimSerializer._write_row(file, serializer, timeserie, config)
+
+    @staticmethod
+    def _write_row(
+        file,
+        serializer: TimTimeSeriesSerializer,
+        timeserie: TimTimeSerie,
+        config: TimSerializerConfig,
+    ):
+        if timeserie.comment:
+            file.write(timeserie.comment)
```

Comment:
The serializer is responsible for putting the `#`/`*` in front of a comment in my opinion. User should be able to set the comment to `blah` and the serializer should write `# blah` in the file.